src/CurrentRoute.php

```diff
@@ -60,9 +110,11 @@ public function getUri(): ?UriInterface
     }
 
     /**
-     * @param RouteParametersInterface $route
+     * @param Route $route
+     *
+     * @internal
      */
-    public function setRoute(RouteParametersInterface $route): void
+    public function setRoute(Route $route): void
```

Comment:
Suggestion: merge methods `setRoute()` and `setArguments` to one method, because in router they used in one place one after the other.